From 1b55ec56b4217bd4bcbf418e512d24534561ed37 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Tue, 4 Feb 2020 11:09:32 +0200
Subject: [PATCH] Move theming to display settings

  * from
  https://github.com/neon-os/packages_apps_Settings/commit/ba07105ed23396f47d78bc3867bcd112ff1584bd

	modified:   res/values/strings.xml
	modified:   res/xml/display_settings.xml
	modified:   src/com/android/settings/DisplaySettings.java
	modified:   src/com/android/settings/development/EmulateDisplayCutoutPreferenceController.java
	renamed:    src/com/android/settings/development/OverlayCategoryPreferenceController.java -> src/com/android/settings/display/OverlayCategoryPreferenceController.java
---
 res/values/strings.xml                                   |  6 ++++++
 res/xml/display_settings.xml                             | 16 ++++++++++++++++
 src/com/android/settings/DisplaySettings.java            |  7 +++++++
 .../EmulateDisplayCutoutPreferenceController.java        |  3 ++-
 .../OverlayCategoryPreferenceController.java             | 16 +++-------------
 5 files changed, 34 insertions(+), 14 deletions(-)
 rename src/com/android/settings/{development => display}/OverlayCategoryPreferenceController.java (92%)

diff --git a/res/values/strings.xml b/res/values/strings.xml
index 65c017e..839c892 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -10048,6 +10048,12 @@
 
     <!-- [CHAR_LIMIT=NONE] Developer Settings: Title of the settings category for theme overlays. -->
     <string name="theme_customization_category">Theming</string>
+    <!-- [CHAR_LIMIT=NONE] Developer Settings: Title of the setting which enables overlays to customize accent color. -->
+    <string name="theme_customization_accent_color_title">Accent color</string>
+    <!-- [CHAR_LIMIT=NONE] Developer Settings: Title of the setting which enables overlays to customize headline and body fonts. -->
+    <string name="theme_customization_font_title">Headline / Body font</string>
+    <!-- [CHAR_LIMIT=NONE] Developer Settings: Title of the setting which enables overlays to customize the adaptive icon shape (e.g. launcher and quick settings icons). -->
+    <string name="theme_customization_icon_shape_title">Icon shape</string>
     <!-- [CHAR_LIMIT=NONE] Developer Settings: Label for the option that turns off customizations for a given category.-->
     <string name="theme_customization_device_default">Device default</string>
 
diff --git a/res/xml/display_settings.xml b/res/xml/display_settings.xml
index d9aad89..81d1ec2 100644
--- a/res/xml/display_settings.xml
+++ b/res/xml/display_settings.xml
@@ -152,4 +152,20 @@
         android:title="@string/display_vr_pref_title"
         android:fragment="com.android.settings.display.VrDisplayPreferencePicker" />
 
+    <PreferenceCategory
+        android:key="theme_customization_category"
+        android:title="@string/theme_customization_category"
+        android:order="1200"> <!-- Incremented by 100 from last order (i.e. 1100) -->
+
+        <ListPreference
+            android:key="android.theme.customization.font"
+            android:title="@string/theme_customization_font_title"/>
+        <ListPreference
+            android:key="android.theme.customization.accent_color"
+            android:title="@string/theme_customization_accent_color_title"/>
+        <ListPreference
+            android:key="android.theme.customization.adaptive_icon_shape"
+            android:title="@string/theme_customization_icon_shape_title"/>
+    </PreferenceCategory>
+
 </PreferenceScreen>
diff --git a/src/com/android/settings/DisplaySettings.java b/src/com/android/settings/DisplaySettings.java
index eb77d4a..ffbb3ff 100644
--- a/src/com/android/settings/DisplaySettings.java
+++ b/src/com/android/settings/DisplaySettings.java
@@ -34,6 +34,7 @@ import com.android.settings.display.TapToWakePreferenceController;
 import com.android.settings.display.ThemePreferenceController;
 import com.android.settings.display.TimeoutPreferenceController;
 import com.android.settings.display.VrDisplayPreferenceController;
+import com.android.settings.display.OverlayCategoryPreferenceController;
 import com.android.settings.search.BaseSearchIndexProvider;
 import com.android.settings.search.Indexable;
 import com.android.settingslib.core.AbstractPreferenceController;
@@ -94,6 +95,12 @@ public class DisplaySettings extends DashboardFragment {
         controllers.add(new ShowOperatorNamePreferenceController(context));
         controllers.add(new ThemePreferenceController(context));
         controllers.add(new BrightnessLevelPreferenceController(context, lifecycle));
+        controllers.add(new OverlayCategoryPreferenceController(context,
+                "android.theme.customization.font"));
+        controllers.add(new OverlayCategoryPreferenceController(context,
+                "android.theme.customization.accent_color"));
+        controllers.add(new OverlayCategoryPreferenceController(context,
+                "android.theme.customization.adaptive_icon_shape"));
         return controllers;
     }
 
diff --git a/src/com/android/settings/development/EmulateDisplayCutoutPreferenceController.java b/src/com/android/settings/development/EmulateDisplayCutoutPreferenceController.java
index ef88baa..c4f7838 100644
--- a/src/com/android/settings/development/EmulateDisplayCutoutPreferenceController.java
+++ b/src/com/android/settings/development/EmulateDisplayCutoutPreferenceController.java
@@ -22,6 +22,7 @@ import android.content.pm.PackageManager;
 import android.os.ServiceManager;
 import android.view.DisplayCutout;
 
+import com.android.settings.display.OverlayCategoryPreferenceController;
 import androidx.annotation.VisibleForTesting;
 
 public class EmulateDisplayCutoutPreferenceController extends OverlayCategoryPreferenceController {
@@ -31,7 +32,7 @@ public class EmulateDisplayCutoutPreferenceController extends OverlayCategoryPre
     @VisibleForTesting
     EmulateDisplayCutoutPreferenceController(Context context, PackageManager packageManager,
             IOverlayManager overlayManager) {
-        super(context, packageManager, overlayManager, DisplayCutout.EMULATION_OVERLAY_CATEGORY);
+        super(context, DisplayCutout.EMULATION_OVERLAY_CATEGORY);
     }
 
     public EmulateDisplayCutoutPreferenceController(Context context) {
diff --git a/src/com/android/settings/development/OverlayCategoryPreferenceController.java b/src/com/android/settings/display/OverlayCategoryPreferenceController.java
similarity index 92%
rename from src/com/android/settings/development/OverlayCategoryPreferenceController.java
rename to src/com/android/settings/display/OverlayCategoryPreferenceController.java
index 6e0b2d0..6a20e3a 100644
--- a/src/com/android/settings/development/OverlayCategoryPreferenceController.java
+++ b/src/com/android/settings/display/OverlayCategoryPreferenceController.java
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package com.android.settings.development;
+package com.android.settings.display;
 
 import static android.os.UserHandle.USER_SYSTEM;
 
@@ -35,8 +35,8 @@ import androidx.preference.Preference;
 import androidx.preference.PreferenceScreen;
 
 import com.android.settings.R;
+import com.android.settingslib.core.AbstractPreferenceController;
 import com.android.settings.core.PreferenceControllerMixin;
-import com.android.settingslib.development.DeveloperOptionsPreferenceController;
 
 import java.util.ArrayList;
 import java.util.Comparator;
@@ -47,7 +47,7 @@ import java.util.List;
  * The chosen overlay is enabled exclusively within its category. A default option is also
  * exposed that disables all overlays in the given category.
  */
-public class OverlayCategoryPreferenceController extends DeveloperOptionsPreferenceController
+public class OverlayCategoryPreferenceController extends AbstractPreferenceController
         implements Preference.OnPreferenceChangeListener, PreferenceControllerMixin {
     private static final String TAG = "OverlayCategoryPC";
     @VisibleForTesting
@@ -194,14 +194,4 @@ public class OverlayCategoryPreferenceController extends DeveloperOptionsPrefere
         filteredInfos.sort(OVERLAY_INFO_COMPARATOR);
         return filteredInfos;
     }
-
-    @Override
-    protected void onDeveloperOptionsSwitchDisabled() {
-        super.onDeveloperOptionsSwitchDisabled();
-        // TODO b/133222035: remove these developer settings when the
-        // Settings.Secure.THEME_CUSTOMIZATION_OVERLAY_PACKAGES setting is used
-        setOverlay(PACKAGE_DEVICE_DEFAULT);
-        updateState(mPreference);
-    }
-
 }
-- 
2.11.0

